e2e/test/utils.ts

```diff
@@ -0,0 +1,33 @@
+import { readdirSync, readFileSync, existsSync } from 'node:fs';
+import { fileURLToPath } from 'node:url';
+
+const TESTS_DIR = fileURLToPath(new URL('../src/content/tutorial/tests', import.meta.url));
```

Comment:
Oh that's a really nice way to get around platform path issues and be more esm idiomatic.